packages/toolpad-app/src/server/localMode.ts

```diff
@@ -816,8 +817,8 @@ export async function openCodeEditor(filePath: string, fileType: string) {
   let resolvedPath = filePath;
 
   if (fileType === 'query') {
-    const resourcesFolder = getResourcesFolder(root);
-    resolvedPath = getFunctionFilePath(resourcesFolder, filePath);
+    const resourcesFolder = await getResourcesFolder(root);
+    resolvedPath = await getFunctionFilePath(resourcesFolder, filePath);
```

Comment:
If you move this whole function into the class you can do

```tsx
resolvedPath = this.functionsManager.getFunctionFilePath(filePath)
```

Reply:
I need to also be able to expose this function to be able to import it in `rpc.ts`, is there a way to do that while also moving this function inside the class?

Reply:
See https://github.com/mui/mui-toolpad/blob/master/packages/toolpad-app/src/server/liveProject.ts